<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ccpmWidget</class>
 <widget class="QWidget" name="ccpmWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>720</width>
    <height>435</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QTabWidget" name="TabObject">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>691</width>
     <height>381</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="tab">
    <attribute name="title">
     <string>Basic settings</string>
    </attribute>
    <widget class="QComboBox" name="ccpmType">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>10</y>
       <width>551</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="toolTip">
      <string>Select aircraft type here</string>
     </property>
     <item>
      <property name="text">
       <string>3 servo 120ยบ - single servo rear</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Custom - Advanced settings</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Swashplate config:</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_3">
    <attribute name="title">
     <string>Curve settings</string>
    </attribute>
    <widget class="QTableWidget" name="CurveSettings">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>190</y>
       <width>261</width>
       <height>141</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="autoScroll">
      <bool>false</bool>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <row>
      <property name="text">
       <string>0%</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>25%</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>50%</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>75%</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>100%</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>none</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>none</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>none</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>none</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>none</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Throttle Curve</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Blade Pitch Curve</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string>0</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="0" column="1">
      <property name="text">
       <string>0</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="1" column="0">
      <property name="text">
       <string>.25</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="1" column="1">
      <property name="text">
       <string>.25</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="2" column="0">
      <property name="text">
       <string>.5</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="2" column="1">
      <property name="text">
       <string>.5</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="3" column="0">
      <property name="text">
       <string>.75</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="3" column="1">
      <property name="text">
       <string>.75</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="4" column="0">
      <property name="text">
       <string>1</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="4" column="1">
      <property name="text">
       <string>1</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="5" column="0">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="5" column="1">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="6" column="0">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="6" column="1">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="7" column="0">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="7" column="1">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="8" column="0">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="8" column="1">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="9" column="0">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="9" column="1">
      <property name="text">
       <string>-</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="CurveType">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>271</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="toolTip">
      <string>Select aircraft type here</string>
     </property>
     <item>
      <property name="text">
       <string>Linear</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Flat</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Step</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Custom</string>
      </property>
     </item>
    </widget>
    <widget class="QSpinBox" name="NumCurvePoints">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>50</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>2</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>50</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Number of points</string>
     </property>
    </widget>
    <widget class="QPushButton" name="ccpmGenerateCurve">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>140</y>
       <width>121</width>
       <height>27</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Generate curves based on settings</string>
     </property>
     <property name="text">
      <string>&lt;-- Generate Curve</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="CurveValue1">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>100</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>1</number>
     </property>
     <property name="maximum">
      <double>10.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="CurveValue2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>100</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>1</number>
     </property>
     <property name="maximum">
      <double>10.000000000000000</double>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="CurveValue3">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>100</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>1</number>
     </property>
     <property name="maximum">
      <double>100.000000000000000</double>
     </property>
     <property name="value">
      <double>50.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="CurveLabel1">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>80</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Min</string>
     </property>
    </widget>
    <widget class="QLabel" name="CurveLabel2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>80</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Max</string>
     </property>
    </widget>
    <widget class="QLabel" name="CurveLabel3">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>80</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Step point</string>
     </property>
    </widget>
    <widget class="QGraphicsView" name="CurveGraph">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>10</y>
       <width>371</width>
       <height>331</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="CurveToGenerate">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>140</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="toolTip">
      <string>Select aircraft type here</string>
     </property>
     <item>
      <property name="text">
       <string>Throttle</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Pitch</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_2">
    <attribute name="title">
     <string>Advanced settings</string>
    </attribute>
    <widget class="QTableWidget" name="ServoSettings">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>251</width>
       <height>141</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="midLineWidth">
      <number>0</number>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="autoScroll">
      <bool>false</bool>
     </property>
     <property name="showDropIndicator" stdset="0">
      <bool>false</bool>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <row>
      <property name="text">
       <string>ServoW</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>ServoX</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>ServoY</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>ServoZ</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Angle</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Output</string>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string>0</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="0" column="1">
      <property name="text">
       <string>8</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="1" column="0">
      <property name="text">
       <string>60</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="1" column="1">
      <property name="text">
       <string>5</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="2" column="0">
      <property name="text">
       <string>180</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="2" column="1">
      <property name="text">
       <string>1</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="3" column="0">
      <property name="text">
       <string>300</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
     <item row="3" column="1">
      <property name="text">
       <string>0</string>
      </property>
      <property name="textAlignment">
       <set>AlignHCenter|AlignVCenter|AlignCenter</set>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>91</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Servo config:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>101</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>CollectiveConstant:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>CorrectionAngle:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>240</y>
       <width>101</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Throttle Output</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>270</y>
       <width>81</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Tail Rotor Servo</string>
     </property>
    </widget>
    <widget class="QComboBox" name="ccpmThrottle">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>240</y>
       <width>151</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="toolTip">
      <string>Select aircraft type here</string>
     </property>
     <property name="currentIndex">
      <number>2</number>
     </property>
     <item>
      <property name="text">
       <string>0</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>3</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>6</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>7</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="ccpmTailRotor">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>270</y>
       <width>151</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="toolTip">
      <string>Select aircraft type here</string>
     </property>
     <property name="currentIndex">
      <number>3</number>
     </property>
     <item>
      <property name="text">
       <string>0</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>3</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>6</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>7</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8</string>
      </property>
     </item>
    </widget>
    <widget class="QDoubleSpinBox" name="CollectiveConstant">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>180</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
     <property name="value">
      <double>0.500000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="CorrectionAngle">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>210</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>0</number>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>15.000000000000000</double>
     </property>
    </widget>
    <widget class="QCheckBox" name="YawStabilizationInManualMode">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>310</y>
       <width>181</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>YawStabilizationInManualMode:</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="Flybarless">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>330</y>
       <width>70</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Flybarless</string>
     </property>
    </widget>
    <widget class="QGraphicsView" name="SwashplateImage">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>10</y>
       <width>371</width>
       <height>331</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QPushButton" name="getccpmCurrent">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>400</y>
     <width>93</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Retrieve settings from OpenPilot</string>
   </property>
   <property name="text">
    <string>Get Current</string>
   </property>
  </widget>
  <widget class="QPushButton" name="saveccpmToRAM">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>400</y>
     <width>93</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Send to OpenPilot but don't write in SD.</string>
   </property>
   <property name="text">
    <string>Save to RAM</string>
   </property>
  </widget>
  <widget class="QPushButton" name="saveccpmToSD">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>400</y>
     <width>93</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Applies and Saves all settings to SD</string>
   </property>
   <property name="text">
    <string>Save to SD</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
